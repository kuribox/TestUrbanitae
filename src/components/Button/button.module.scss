$primaryColor: #88b04b;
$secondaryColor: white;

.label {
    color: $secondaryColor;
    text-transform: uppercase;
}

.container {
    background-color: $primaryColor;
    height: 30px;
    border-radius: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 8px;
    min-width: 100px;
    cursor: pointer;
    border: 2px solid $primaryColor;

    &:hover {
        background-color: $secondaryColor;
        & .label {
            color: $primaryColor;
        }
    }
}

.labelAlt {
    color: $primaryColor;
}

.containerAlt {
    background-color: $secondaryColor;
    border: 2px solid $primaryColor;

    &:hover {
        background-color: $primaryColor;
        & .label {
            color: $secondaryColor;
        }
    }
}
