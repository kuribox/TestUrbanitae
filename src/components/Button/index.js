import React from 'react';
import styles from './button.module.scss';
import cn from 'classnames';

const Button = ({ label, alternative, ...props}) => {
  return (
    <>
      <div className={cn([styles.container, alternative ? styles.containerAlt : null])} {...props}>
        <span className={cn([styles.label, alternative ? styles.labelAlt : null])}>{label}</span>
      </div>
    </>
  );
}

export default Button;