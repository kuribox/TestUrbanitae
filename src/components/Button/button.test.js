import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';
import Button from './index.js';

test('Snapshot check', () => {
  const component = renderer.create(
    <Button />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Check Label', () => {
  const { getByText } = render(
    <Button label="test" />,
  );

  expect(getByText("test")).toBeInTheDocument();
});

test('Check click', () => {
  let button = false;
  const component = renderer.create(
    <Button onClick={() => {
      button = true;
    }}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();

  tree = component.toJSON();
  expect(button).toBe(true);
});
