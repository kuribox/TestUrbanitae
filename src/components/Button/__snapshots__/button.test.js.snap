// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check click 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <span
    className="label"
  />
</div>
`;

exports[`Snapshot check 1`] = `
<div
  className="container"
>
  <span
    className="label"
  />
</div>
`;
